pkgs/applications/editors/neovim/qt.nix

```diff
@@ -1,62 +1,74 @@
 { stdenv, fetchFromGitHub, cmake, doxygen, makeWrapper
 , msgpack, neovim, pythonPackages, qtbase }:
 
-stdenv.mkDerivation rec {
-  name = "neovim-qt-${version}";
-  version = "0.2.11";
-
-  src = fetchFromGitHub {
-    owner  = "equalsraf";
-    repo   = "neovim-qt";
-    rev    = "v${version}";
-    sha256 = "0pc1adxc89p2rdvb6nxyqr9sjzqz9zw2dg7a4ardxsl3a8jga1wh";
-  };
+let
+  unwrapped = stdenv.mkDerivation rec {
+    pname = "neovim-qt-unwrapped";
+    version = "0.2.11";
+
+    src = fetchFromGitHub {
+      owner  = "equalsraf";
+      repo   = "neovim-qt";
+      rev    = "v${version}";
+      sha256 = "0pc1adxc89p2rdvb6nxyqr9sjzqz9zw2dg7a4ardxsl3a8jga1wh";
+    };
+
+    cmakeFlags = [
+      "-DUSE_SYSTEM_MSGPACK=1"
+    ];
+
+    buildInputs = [
+      neovim.unwrapped qtbase
+    ] ++ (with pythonPackages; [
+      jinja2 python msgpack
+    ]);
+
+    nativeBuildInputs = [ cmake doxygen makeWrapper ];
+
+    enableParallelBuilding = true;
+
+    preCheck = ''
+      # The GUI tests don't work, retry when
+      # https://github.com/NixOS/nixpkgs/pull/44047 is resolved.
+      sed -i ../test/CMakeLists.txt \
+        -e '/^add_xtest_gui/d'
```

Comment:
You might be able to run this with:

```
export QT_PLUGIN_PATH ${qtbase}/${qtbase.qtPluginPrefix}
```
as a stop-gap solution until we get the global fix in https://github.com/NixOS/nixpkgs/pull/44047

Reply:
I tried that, it didn't fix the issue:

```
4/7 Test #7: tst_shell ........................***Failed    0.09 sec
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/build/runtime-nixbld'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
```

Reply:
Interesting, that suggests it's unlikely to work even with #44047, since it requires not just QT plugins but a full X display.

Reply:
You're right. I've amended that comment.